### DISCUSSION TITLE
ESP32 Sketch size at 99%

### DISCUSSION THREAD

**Discussion by greymda** [hoylabs/OpenDTU-OnBattery#896] [General]
<img width="1318" alt="image" src="https://github.com/helgeerbe/OpenDTU-OnBattery/assets/9991596/7f2e5167-62db-477c-a377-7f57d3bbd644">

Is this by design or it's a local problem only for me?

Firmware Version / Git Hash	[2024.03.23]
Firmware Branch	master
PIO Environment	generic_esp32

**Comment by Manos1966** [comment]
Made by design.

and no, you are NOT the first one who jumped 😄 😄 😄 😄

**Comment by schlimmchen** [comment]
There are discussions buried deep in issued and pull request revolving around this. This limitation is known, and addressing it will be a pain. The current partition layout is for 4MB flash, and it contains two full firmware images. We will try to save as much flash as possible to postpone having to address this issue as long as possible. Do you need to know more?

**Comment by greymda** [comment]
@schlimmchen i had a couple of resets and wondered the reason

 Reset Reason CPU 0	Software reset CPU
Reset Reason CPU 1	Software reset CPU

but. most probably unrelated as i see you’re aware and okay with this.

**Reply by schlimmchen** [reply]
That is annoying. If you can, capture a serial log while the unwanted reboots happen, then we can have a look at it. There should be a backtrace which should give us a good idea about where the problem is. Open a new issue if you have the log.

**Reply by spcqike** [reply]
I had a similar issue. Mentioned here 
https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/700#discussioncomment-8619919

I also had several „software resets“ but it looks like it was related to the power supply. As the problems disappeared after I usedanother power supply 

at least the resets where fewer afterwards
